extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java

```diff
@@ -381,8 +377,10 @@ protected void containerIsStarting(InspectContainerResponse containerInfo, boole
             command += "/usr/bin/rpk redpanda start --check=false --node-id 0 --smp 1 ";
             command += "--memory 1G --overprovisioned --reserve-memory 0M ";
             command += "--kafka-addr PLAINTEXT://0.0.0.0:29092,OUTSIDE://0.0.0.0:9092 ";
-            command += String.format("--advertise-kafka-addr PLAINTEXT://%s:29092,OUTSIDE://%s:%d", getHostToUse(),
-                    getHostToUse(), getPortToUse());
+            command += String.format("--advertise-kafka-addr PLAINTEXT://%s:29092,OUTSIDE://%s:%d ",
```

Comment:
Form what I can gather (I am not a Kafka expert by any stretch of anyone's imagination) we always need to define a listener for the "local docker network" and hence need the host name that will be available on the bridge. See https://rmoff.net/2018/08/02/kafka-listeners-explained/ that guided my change (and allows my environment to work).

Reply:
@ozangunalp cab you review this part? You did the network configuration.

Reply:
That seems correct, the broker will be accessible on `hostname:29092` from inside the container's network.
And +1 for the article, it's a go-to for Kafka listeners config.